File: CS/DevExpressWebGardenReporting/CustomWebDocumentViewerOperationLogger.cs
using System;
using DevExpress.XtraPrinting;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.ClientControls;
using DevExpress.XtraReports.Web.WebDocumentViewer;

namespace DevExpressWebGardenReporting {
    public class CustomWebDocumentViewerOperationLogger : WebDocumentViewerOperationLogger {
        public override Action BuildStarting(string reportId, string reportUrl, XtraReport report, ReportBuildProperties buildProperties) {
            report.CreateDocument();
            return base.BuildStarting(reportId, reportUrl, report, buildProperties);
        }
        public override ExportedDocument ExportDocumentStarting(string documentId, string asyncExportOperationId, string format, ExportOptions options, PrintingSystemBase printingSystem, Func<ExportedDocument> doExportSynchronously) {
            return doExportSynchronously();
        }
    }
}

File: VB/DevExpressWebGardenReporting/CustomWebDocumentViewerOperationLogger.vb
Imports System
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.ClientControls
Imports DevExpress.XtraReports.Web.WebDocumentViewer

Namespace DevExpressWebGardenReporting

    Public Class CustomWebDocumentViewerOperationLogger
        Inherits WebDocumentViewerOperationLogger

        Public Overrides Function BuildStarting(ByVal reportId As String, ByVal reportUrl As String, ByVal report As XtraReport, ByVal buildProperties As ReportBuildProperties) As Action
            report.CreateDocument()
            Return MyBase.BuildStarting(reportId, reportUrl, report, buildProperties)
        End Function

        Public Overrides Function ExportDocumentStarting(ByVal documentId As String, ByVal asyncExportOperationId As String, ByVal format As String, ByVal options As ExportOptions, ByVal printingSystem As PrintingSystemBase, ByVal doExportSynchronously As Func(Of ExportedDocument)) As ExportedDocument
            Return doExportSynchronously()
        End Function
    End Class
End Namespace
